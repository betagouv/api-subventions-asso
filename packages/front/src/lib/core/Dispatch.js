import { createEventDispatcher } from "svelte";

export default class Dispatch {
    static getDispatcher = () => createEventDispatcher();
}
