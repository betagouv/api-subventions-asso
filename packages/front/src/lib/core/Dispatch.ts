import { createEventDispatcher } from "svelte";

export default class Dispatch {
    static getDispatcher = <T>() => createEventDispatcher<Record<string, T>>();
}
