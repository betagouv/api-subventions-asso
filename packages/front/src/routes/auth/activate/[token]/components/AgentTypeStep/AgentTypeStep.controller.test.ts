import { beforeEach } from "vitest";
import AgentTypeStepController from "./AgentTypeStep.controller";
import Dispatch from "$lib/core/Dispatch";

vi.mock("$lib/core/Dispatch", () => ({
    default: {
        getDispatcher: vi.fn(() => vi.fn()),
    },
}));
vi.mock("$lib/core/Store", async () => {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any -- mock
    const actual = (await vi.importActual("$lib/core/Store")) as any;
    return {
        ...actual,
        derived: vi.fn(actual.derived),
    };
});
vi.mock("svelte");

describe("AgentTypeStepController", () => {
    let ctrl: AgentTypeStepController;

    beforeEach(() => {
        ctrl = new AgentTypeStepController();
    });

    describe("constructor", () => {
        it("inits dispatch", () => {
            const expected = "Result From Dispatch";
            // @ts-expect-error - mock
            vi.mocked(Dispatch.getDispatcher).mockReturnValueOnce(expected);
            ctrl = new AgentTypeStepController();
            // @ts-expect-error - mock private
            const actual = ctrl.dispatch;
            expect(actual).toBe(expected);
        });

        it("inits errorMessage", () => {
            const expected = "";
            const actual = ctrl.errorMessage.value;
            expect(actual).toBe(expected);
        });

        it("inits options", () => {
            ctrl = new AgentTypeStepController();
            const actual = ctrl.options;
            expect(actual).toMatchSnapshot();
        });
    });

    describe.each`
        withError | event      | option
        ${true}   | ${"error"} | ${{ value: "none" }}
        ${false}  | ${"valid"} | ${{ value: "some" }}
    `("onUpdate - case $event", ({ event, option }) => {
        it("updates errorMessage store", () => {
            // @ts-expect-error - mock readonly
            ctrl.errorMessage = { set: vi.fn() };
            ctrl.onUpdate(option);
            expect(ctrl.errorMessage.set).toMatchSnapshot();
        });

        it("dispatches event", () => {
            // @ts-expect-error - mock private
            ctrl.dispatch = vi.fn();
            ctrl.onUpdate(option);
            // @ts-expect-error - test private
            expect(ctrl.dispatch).toHaveBeenCalledWith(event);
        });
    });
});
