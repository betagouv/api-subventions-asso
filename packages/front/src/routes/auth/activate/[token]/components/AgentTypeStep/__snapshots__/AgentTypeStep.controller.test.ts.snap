// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AgentTypeStepController > constructor > inits options 1`] = `
[
  {
    "label": "Agent public d’une administration centrale (État)",
    "value": "CENTRAL_ADMIN",
  },
  {
    "label": "Agent public d’une administration déconcentrée (État)",
    "value": "DECONCENTRATED",
  },
  {
    "label": "Agent public d’une collectivité territoriale",
    "value": "TERRITORIAL_COLLECTIVITY",
  },
  {
    "label": "Agent public d’un opérateur de l’État",
    "value": "OPERATOR",
  },
  {
    "label": "Aucune des propositions ci-dessus",
    "value": "none",
  },
]
`;

exports[`AgentTypeStepController > onUpdate - case 'error' > updates errorMessage store 1`] = `
[MockFunction spy] {
  "calls": [
    [
      "Data.Subvention est réservé aux agents publics. Pour toute question, vous pouvez nous contacter à <a href=\\"mailto:contact@datasubvention.beta.gouv.fr\\">contact@datasubvention.beta.gouv.fr</a>",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`AgentTypeStepController > onUpdate - case 'valid' > updates errorMessage store 1`] = `
[MockFunction spy] {
  "calls": [
    [
      "",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
